MetadataExtractor/Formats/Jpeg/JpegMetadataReader.cs

```diff
@@ -114,7 +116,10 @@ public static DirectoryList ProcessJpegSegments([NotNull] IEnumerable<IJpegSegme
             {
                 var readerSegmentTypes = reader.SegmentTypes;
                 var readerSegments = segments.Where(s => readerSegmentTypes.Contains(s.Type));
-                directories.AddRange(reader.ReadJpegSegments(readerSegments));
+                if(reader is JpegDnlReader)
+                    ((JpegDnlReader)reader).ReadJpegSegments(readerSegments, directories);
```

Comment:
This can use pattern matching:

```c#
                if (reader is JpegDnlReader jpegDnlReader)
                    jpegDnlReader.ReadJpegSegments(readerSegments, directories);
```

However I'm not sure I like this pattern. I'll comment on `JpegDnlReader` explaining why.